src/features/common/RedeemCode/EnterRedeemCode.tsx

```diff
@@ -0,0 +1,65 @@
+import TextField from '@mui/material/TextField';
+import { useTranslation } from 'next-i18next';
+import { useForm } from 'react-hook-form';
+import { ReactElement } from 'react';
+import CancelIcon from '../../../../public/assets/images/icons/CancelIcon';
+import styles from '../../../../src/features/user/Profile/styles/RedeemModal.module.scss';
+import Button from '@mui/material/Button';
+
+export interface EnterRedeemCodeProps {
+  setInputCode: React.Dispatch<React.SetStateAction<string | null>>;
+  inputCode: string | null;
+  changeRouteCode: () => void;
+  closeRedeem: () => void;
+}
+
+const EnterRedeemCode = ({
+  setInputCode,
+  inputCode,
+  changeRouteCode,
+  closeRedeem,
+}: EnterRedeemCodeProps): ReactElement => {
+  const { register, errors, handleSubmit } = useForm({ mode: 'onBlur' });
+  const { t } = useTranslation(['redeem']);
+
+  return (
+    <div className={styles.routeRedeemModal}>
+      <div className={styles.crossDiv}>
+        <button className={styles.crossWidth} onClick={closeRedeem}>
+          <CancelIcon />
+        </button>
+      </div>
+
+      <div className={styles.redeemHeading}>{t('redeem:redeem')}</div>
+      <div className={styles.note}>{t('redeem:redeemDescription')}</div>
+      <div className={styles.inputField}>
+        <TextField
+          inputRef={register({
+            required: {
+              value: true,
+              message: t('redeem:enterRedeemCode'),
+            },
+          })}
+          onChange={(event) => {
+            setInputCode(event.target.value);
+          }}
+          value={inputCode}
+          name={'code'}
+          placeholder="XAD-1SA-5F1-A"
+          label=""
+          variant="outlined"
+          error={errors.code}
+          helperText={errors.code && errors.code.message}
+        />
+      </div>
+
+      <div>
+        <Button variant="contained" onClick={handleSubmit(changeRouteCode)}>
+          {t('redeem:redeemCode')}
+        </Button>
```

Comment:
There should be some loading state

Reply:
to implement the loading state there is need to create state variable for it. So once this [https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/1707/commits/78399f602c1b51f93cbe1a28799af135eb70f771](PR) get merge into develop I can get a state variable (`disable`) which is there in userPropContext

Reply:
Why would you use a variable from userPropContext for disabling the action? 

It seems more suitable to use a local state variable, which gets before and after the redeem code api call. Using context indiscriminately makes the code tougher to read, and can also lead to conflicts (for example, if one component updates the value of `disable` while it's in use somewhere else).

Use context when we need to store data that is used for multiple components, and pass it through multiple levels of a component tree.

Reply:
Minor adjustments made here -  032160ef1253396972a4677a31d46fc5a58aaa71